packages/usa-input-mask/src/index.js

```diff
@@ -1,124 +1,1091 @@
 const selectOrMatches = require("../../uswds-core/src/js/utils/select-or-matches");
 const behavior = require("../../uswds-core/src/js/utils/behavior");
+const debounce = require("../../uswds-core/src/js/utils/debounce");
 const { prefix: PREFIX } = require("../../uswds-core/src/js/config");
 
-const MASKED_CLASS = `${PREFIX}-masked`;
+const MASKED_CLASS = `${PREFIX}-input-mask`;
 const MASKED = `.${MASKED_CLASS}`;
-const MASK = `${PREFIX}-input-mask`;
-const MASK_CONTENT = `${MASK}--content`;
+const MASK_SHELL = `${MASKED_CLASS}__shell`;
+const MASK_CONTENT = `${MASKED_CLASS}__content`;
+const INPUT = `${MASKED}__field`;
+const MESSAGE = `${MASKED}__message`;
 const PLACEHOLDER = "placeholder";
-const CONTEXT = "form";
-
-// User defined Values
-const maskedNumber = "_#dDmMyY9";
-const maskedLetter = "A";
-
-// replaces each masked input with a shell containing the input and it's mask.
-const createMaskedInputShell = (input) => {
-  const placeholder = input.getAttribute(`${PLACEHOLDER}`);
-  if (placeholder) {
-    input.setAttribute("maxlength", placeholder.length);
-    input.setAttribute("data-placeholder", placeholder);
-    input.removeAttribute(`${PLACEHOLDER}`);
+
+const MESSAGE_INVALID_CLASS = `${MASKED_CLASS}__status--invalid`;
+const STATUS_MESSAGE_CLASS = `${MASKED_CLASS}__status`;
+const STATUS_MESSAGE_SR_ONLY_CLASS = `${MASKED_CLASS}__sr-status`;
+const STATUS_MESSAGE = `.${STATUS_MESSAGE_CLASS}`;
+const STATUS_MESSAGE_SR_ONLY = `.${STATUS_MESSAGE_SR_ONLY_CLASS}`;
+const DEFAULT_STATUS_LABEL = `character is required here`;
```

Comment:
Should this be a data attribute that the user can set in case they want to use another language?

Reply:
I followed the character count paradigm for this which has hardcoded English messages. I think a data attribute would be good though! https://github.com/uswds/uswds/blob/develop/packages/usa-character-count/src/index.js